cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -98,6 +110,42 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
     }
   }
 
+  /**
+   * Returns the Git file hash of the requested file path in the current HEAD.
+   *
+   * @param path The absolute path on the filesystem that should include {@link RepositoryManager#getBasePath()} as a
+   *             prefix.
+   * @return The git file hash of the requested file.
+   * @throws IOException              when file isn't found or file isn't committed to Git.
+   * @throws IllegalArgumentException when the provided file isn't a regular file, eg: symlink or directories.
+   * @throws NotFoundException        when the file isn't committed to Git.
+   */
+  public String getFileHash(Path path) throws IOException, NotFoundException {
+    validateInitialized();
+    Path repoBasePath = this.getBasePath();
+    if (!path.startsWith(repoBasePath) || !Files.exists(path, NOFOLLOW_LINKS)) {
```

Comment:
Why are we not following links here ? 
There can be valid usecases where the user wants to symlink to existing config files to align with naming convention that cdap wants to follow.

Reply:
For git, even a symlink is a file, so the the hash of the symlink will be different from the hash of the actual file. To overcome this, we need to follow the symlink to find the path to the actual file.
Is it fine to follow a single level of link, i.e a->b or should we follow multiple links till we find a regular file (a->b->c->d...)?

Reply:
Discussed offline, we will use `Paths#toRealPath()` that handles multi level links and also throws an exception when it encounters circular links.

Reply:
Done.